fauna/client/client.py

```diff
@@ -574,6 +638,82 @@ def close(self):
       self._stream.close()
 
 
+@dataclass
+class ChangeFeedPage:
+  events: List[Any]
+  cursor: str
+  stats: QueryStats
+
+  def __iter__(self) -> Iterator[Any]:
+    return iter(self.events)
+
+
+class ChangeFeedIterator:
+  """A class to provide an iterator on top of change feed pages."""
+
+  def __init__(self, http: HTTPClient, headers: Dict[str, str],
+               endpoint: str, max_attempts: int, max_backoff: int,
+               opts: ChangeFeedOptions, token: StreamToken):
+    self._http = http
+    self._headers = headers
+    self._endpoint = endpoint
+    self._max_attempts = opts.max_attempts or max_attempts
+    self._max_backoff = opts.max_backoff or max_backoff
+    self._request = {"token": token.token}
+    self._is_done = False
+
+    if opts.page_size is not None:
+      self._request["page_size"] = opts.page_size
+
+    if opts.cursor is not None:
+      self._request["cursor"] = opts.cursor
+    elif opts.start_ts is not None:
+      self._request["start_ts"] = opts.start_ts
+
+  def __iter__(self) -> Iterator[ChangeFeedPage]:
+    self._is_done = False
+    return self
+
+  def __next__(self) -> ChangeFeedPage:
+    if self._is_done:
+      raise StopIteration
+
+    retryable = Retryable[Any](self._max_attempts, self._max_backoff, self._next_page)
+    return retryable.run().response
+
+  def _next_page(self) -> ChangeFeedPage:
+    with self._http.request(
+        method="POST",
+        url=self._endpoint,
+        headers=self._headers,
+        data=self._request,
+    ) as response:
+      status_code = response.status_code()
+      decoded = FaunaDecoder.decode(response.json())
+
+      if status_code > 399:
+        FaunaError.parse_error_and_throw(decoded, status_code)
+
+      self._is_done = not decoded["has_next"]
+      self._request["cursor"] = decoded["cursor"]
+
+      if "start_ts" in self._request:
+        del self._request["start_ts"]
+
+      for event in decoded["events"]:
+        if event["type"] == "error":
+          FaunaError.parse_error_and_throw(event, 400)
```

Comment:
Is it actually possible for some change feed events to be errors? Why would an individual event be an error, rather than the whole paged response?

Reply:
I've gone back and forth on this one. The behavior that I think we've setting is: the page will stop at first error; you can ignore it and pull the next page to continue the feed. I've changed the interface to support the desired behavior and I'll follow up with a PR on the core side to implement it.